lib/friendly_shipping/services/ship_engine.rb

```diff
@@ -15,36 +15,47 @@ class ShipEngine
         labels: "labels"
       }
 
-      def initialize(token:, test: true)
+      def initialize(token:, test: true, client: self.class::Client)
         @token = token
         @test = test
+        @client = client
       end
 
       def carriers
-        path = API_BASE + API_PATHS[:carriers]
-        FriendlyShipping::RestClient.get(path, request_headers).fmap do |response|
+        request = FriendlyShipping::Request.new(
+          url: API_BASE + API_PATHS[:carriers],
+          headers: request_headers
+        )
+        client.get(request).fmap do |response|
           ParseCarrierResponse.new(response: response).call
         end
       end
 
       def labels(shipment)
-        payload = SerializeLabelShipment.new(shipment: shipment).call.merge(test_label: test).to_json
-        path = API_BASE + API_PATHS[:labels]
-        FriendlyShipping::RestClient.post(path, payload, request_headers).fmap do |response|
-          ParseLabelResponse.new(response: response).call
+        request = FriendlyShipping::Request.new(
+          url: API_BASE + API_PATHS[:labels],
+          body: SerializeLabelShipment.new(shipment: shipment).call.merge(test_label: test).to_json,
+          headers: request_headers
+        )
+        client.post(request).fmap do |response|
+          ParseLabelResponse.(request: request, response: response)
         end
       end
 
       def void(label)
-        path = "#{API_BASE}labels/#{label.id}/void"
-        FriendlyShipping::RestClient.put(path, '', request_headers).bind do |response|
+        request = FriendlyShipping::Request.new(
+          url: "#{API_BASE}labels/#{label.id}/void",
+          body: '',
+          headers: request_headers
+        )
+        client.put(request).bind do |response|
           ParseVoidResponse.new(response: response).call
```

Comment:
Again, for consistency, shouldn't this be `ParseVoidResponse.(response: response)`?